invokeai/app/services/generate_initializer.py

```diff
@@ -1,16 +1,17 @@
 import os
 import sys
+import torch
 import traceback
 from argparse import Namespace
+from omegaconf import OmegaConf
 
 import invokeai.version
-from invokeai.backend import Generate, ModelManager
-
+from ...backend import ModelManager
+from ...backend.util import choose_precision, choose_torch_device
 from ...backend import Globals
 
-
 # TODO: most of this code should be split into individual services as the Generate.py code is deprecated
-def get_generate(args, config) -> Generate:
+def get_model_manager(args, config) -> ModelManager:
     if not args.conf:
```

Comment:
I *think* OmegaConf can do all this (layer config files under command line args). We should probably split out configuration loading somewhere separate, and pass strongly-typed configs to construct classes like this.

Reply:
Thanks for this comment. I went back and looked at how `cli_app.py` is getting and passing around the command-line arguments. I've fixed the argument parsing in `cli_app.py`, `api_app.py` and `dependencies.py`.

We don't actually have a YAML-style configuration file at the moment. `invokeai.init` is a argparse-format  file that is loaded when you create the `Args` object, and merging of options defined in the init file are already merged with command-line arguments internally.

Reply:
Ah. Both of the _app files are fairly hacky around startup, since I figured we'd clean them up later 🙂.